libraries/botbuilder-runtime/src/configuration.ts

```diff
@@ -97,6 +108,31 @@ export class Configuration implements IConfiguration {
         return this.type(path, String);
     }
 
+    /**
+     * Get a int value from config
+     *
+     * @param path path to numeric value
+     * @returns the number or undefined
+     */
+    async integer(path: string[]): Promise<number | undefined> {
+        const runtype = Union(Number, String);
+        const value = await this.type(path, runtype);
+        if (value) {
+            return runtype.match(
+                (num) => num,
+                (str) => {
+                    const parsed = parseInt(str, 10);
+
+                    if (isNaN(parsed)) {
+                        throw new Error(`${str} is not an integer`);
```

Comment:
How do you feel about throwing a `TypeError`? Do you have any stance on the SDK throwing the appropriate error vs the generic `Error`?

Reply:
Yep - an oversight on my part! Will fix.